### PR TITLE
Use fetched cipher/digest for tickets

### PR DESCRIPTION
**Pull request by mattcaswell**
Make sure we use explicitly fetched cipher and digest when handling libssl tickets.

This is needed for TLS using only FIPS and contains commits from #11334

### CODE REVIEW

File: crypto/err/openssl.txt [hunk 1]
```diff
@@ -3088,6 +3088,7 @@ SSL_R_EXTENSION_NOT_RECEIVED:279:extension not received
 SSL_R_EXTRA_DATA_IN_MESSAGE:153:extra data in message
 SSL_R_EXT_LENGTH_MISMATCH:163:ext length mismatch
 SSL_R_FAILED_TO_INIT_ASYNC:405:failed to init async
+SSL_R_FETCH_FAILED:295:fetch failed
```
Comment (t8m) [inline crypto/err/openssl.txt]: Could the error be a little more descriptive? Something like "algorithm fetch failed" or "evp algorithm fetch failed".

File: ssl/statem/statem_srvr.c [hunk 2]
```diff
@@ -3906,7 +3906,14 @@ static int construct_stateless_ticket(SSL *s, WPACKET *pkt, uint32_t age_add,
         }
         iv_len = EVP_CIPHER_CTX_iv_length(ctx);
     } else {
-        const EVP_CIPHER *cipher = EVP_aes_256_cbc();
+        EVP_CIPHER *cipher = EVP_CIPHER_fetch(s->ctx->libctx, "AES-256-CBC",
+                                              s->ctx->propq);
+
+        if (cipher == NULL) {
+            SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_CONSTRUCT_STATELESS_TICKET,
+                     ERR_R_INTERNAL_ERROR);
```
Comment (t8m) [inline ssl/statem/statem_srvr.c]: And it could be used here as well?

File: crypto/err/openssl.txt (hunk 1 above)
Comment (mattcaswell) [reply crypto/err/openssl.txt]: I made it "algorithm fetch failed"

File: ssl/statem/statem_srvr.c (hunk 2 above)
Comment (mattcaswell) [reply ssl/statem/statem_srvr.c]: Done

**Comment by mattcaswell** [comment]
Please take another look

**Comment by t8m** [comment]
Actually the Travis failure - a leak seems to be relevant.

**Comment by t8m** [comment]
There is one return where the fetched sha256 is not freed.

**Comment by mattcaswell** [comment]
Fixup pushed addressing the memory leak.

@t8m - please take another look.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

Matt

**Pull request closed without merging**